src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java

```diff
@@ -126,18 +128,28 @@ public Page<AdvertisementTitleDto> findAdvertisementsHavingCategory(
             @RequestParam(value = "size", required = false, defaultValue = "12") @PositiveOrZero int size) throws CategoryIdNotFoundException {
         if (!categoryService.isCategoryExistsById(categoryId)) {
             throw new CategoryIdNotFoundException(
-                    getParametrizedMessageSource(ResponseMessagesHandler.ValidationMessage.INVALID_CATEGORY_ID, categoryId));
+                    getParametrizedMessageSource(INVALID_CATEGORY_ID, categoryId));
         }
         return advertisementService.findByCategoryId(categoryId, PageRequest.of(page, size));
     }
 
-    @PostMapping(value = ApiKey.ADV_FILTER, produces = MediaType.APPLICATION_JSON_VALUE)
-    @Operation(summary = "Filter advertisements by multiple params and return up to 10 results.\n" +
-            "Fill only needed parameters.")
+    @GetMapping(value = ApiKey.ADV_FILTER, produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "Filter advertisements by multiple params")
     @ApiResponse(responseCode = "200", description = "OK")
     @ResponseStatus(HttpStatus.OK)
-    public List<AdvertisementTitleDto> getFirst10BySearchParameters(@Valid @RequestBody AdvertisementFilterDto filterDto) {
-        return advertisementService.findFirst10ByFilter(filterDto);
+    public Page<AdvertisementTitleDto> filterAdvertisementBySearchParameters(
+            @Valid @ParameterObject AdvertisementFilterRequest filterRequest) {
+        validateAllSubcategoriesExistsInCategory(filterRequest.getSubcategorySearchRequest());
+        return advertisementService.filterAdvertisementBySearchParameters(filterRequest);
+    }
+
+    private void validateAllSubcategoriesExistsInCategory(SubcategorySearch subcategorySearch) {
+        List<Long> searchSubcategoriesIdValues = subcategorySearch.getSubcategoriesIdValues();
+        List<Long> existingSubcategoriesId = subcategoryService.findExistingIdForCategoryById(
+                subcategorySearch.getCategoryId(), searchSubcategoriesIdValues);
+        if (existingSubcategoriesId.size() != searchSubcategoriesIdValues.size()) {
+            throw new BadRequestException(getMessageSource(INVALID_CATEGORY_SUBCATEGORY_COMBINATION));
```

Comment:
You should pass in the exception message the list of subcategories ID which only were found including the details of the main exception cause